pkg/console/controllers/util/util.go

```diff
@@ -57,3 +59,28 @@ func LabelFilter(labels map[string]string) factory.EventFilterFunc {
 		return true
 	}
 }
+
+type consoleOperatorController interface {
+	HandleManaged(context.Context) error
+	HandleUnmanaged(context.Context) error
+	HandleRemoved(context.Context) error
+	GetOperatorSpec() (*operatorv1.OperatorSpec, error)
+}
+
+func HandleManagementState(ctx context.Context, c consoleOperatorController) error {
+	operatorSpec, err := c.GetOperatorSpec()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve operator config: %w", err)
+	}
+
+	switch managementState := operatorSpec.ManagementState; managementState {
```

Comment:
just a note, no action necessary:
I've always thought that if I'm going to do it this, I'd keep the handled conditions in a map, so that I can do this:
```go
handler, ok := c.stateHandlers[managementState]
if !ok {
    return fmt.Errorf("console is in an unknown state: %v", managementState)
}

// noop handling a state
if handler == nil {
    klog.V(4).Infof("handling %q as noop", managementState)
    return nil
}

return handler(ctx)
```

I find it a bit more elegant than the interfaces but it's just a personal preference.